src/Tokenizer.js

```diff
@@ -12,11 +12,12 @@ export var characters = ''
  * @param {object} root
  * @param {string} type
  * @param {string[]} props
+ * @param {object|null} parent
  * @param {object[]} children
  * @param {number} length
  */
-export function node (value, root, type, props, children, length) {
-	return {value: value, root: root, type: type, props: props, children: children, line: line, column: column, length: length, return: ''}
+export function node (value, root, type, props, parent, children, length) {
+	return {value: value, root: root, type: type, props: props, parent: parent, children: children, line: line, column: column, length: length, return: ''}
```

Comment:
I'm sentimental about `type`, `props`, `children` always being one after another.

Reply:
Adjusted all function to always accept `parent` after the `root`